sdlib/d/gc/slab.d

```diff
@@ -223,12 +224,15 @@ public:
 		       "New finalizer pointer is invalid!");
 
 		auto freeSpaceValue = slotSize - initialUsedCapacity;
-		bool isLarge = freeSpaceValue > 0x3f;
-		auto finalizerSet = hasFinalizer << 1;
-		ushort native =
-			(freeSpaceValue << 2 | isLarge | finalizerSet) & ushort.max;
+		ushort native = void;
+		if (freeSpaceValue == 1) {
+			native = SingleByteBit;
+		} else {
+			auto finalizerSet = hasFinalizer ? FreeSpaceFinalizerBit : 0;
```

Comment:
You only set the finalizer bit when the size is not 1.

The fact this can be missed also show it warrants a test case.

Reply:
There is an assert that prevents this up above.

```d
 assert(
                        initialUsedCapacity <= slotSize - (hasFinalizer ? PointerSize : 0),
                        "Insufficient alloc capacity!"
                );
```
i.e. if a finalizer is set, then the free space must be at least PointerSize bytes.

I can add a second assert for clarity.